CondTools/SiPixel/plugins/SiPixel2DTemplateDBObjectUploader.cc

```diff
@@ -90,8 +89,7 @@ void SiPixel2DTemplateDBObjectUploader::analyze(const edm::Event& iEvent, const
   // Open the template file(s)
   for (m = 0; m < obj.numOfTempl(); ++m) {
     edm::FileInPath file(theTemplateCalibrations[m].c_str());
-    tempfile = (file.fullPath()).c_str();
-    std::ifstream in_file(tempfile, std::ios::in);
+    std::ifstream in_file(file.fullPath().c_str(), std::ios::in);
```

Comment:
IIRC the language rules correctly, this is not sufficient, because it is only the object returned by `c_str()` that is guaranteed to be alive throughout the `ifstream` constructor call, but not the intermediate (temporary) object returned by `fullPath()`.

Reply:
It passes though

Reply:
Right, I was wrong, the temporary returned by `fullPath()` is guaranteed to live until the constructor returns (technically "temporary objects are destroyed as the last step in evaluating the full-expression"). I mixed this to the use of intermediate temporaries in range-for. Sorry for the noise.